### PR TITLE
clean utm_term url parameter

### PR DESCRIPTION
**Pull request by nodiscc**
Reported in https://github.com/shaarli/Shaarli/pull/141#issuecomment-77268039

### CODE REVIEW

**Comment by Marsup** [comment]
Why not utm_* ?

**Comment by virtualtam** [comment]
+1 for a generic `utm_crap=more-crap` remover
```php
/[\?&]utm_[^&]*=[^&]*/
```

Btw, the URL cleaning function could be moved to a new place (e.g. `application/Utils.php` or `application/Url.php`) & test-covered :)

**Comment by nodiscc** [comment]
Can't this be simplified to

```php
    /[\?&]utm_[^&]*/
```

?

**Comment by virtualtam** [comment]
I guess so :)

**Comment by ArthurHoaro** [comment]
Also there is a use case not handled:

  http://link.com?utm_key=stuff&otherparam=1

Will become:

  http://link.com&otherparam=1

Which is incorrect and happened in @nicolasdanelon report.

**Comment by virtualtam** [comment]
We could:
- build an array of all GET parameters present in the URL
- remove all entries matching _annoying patterns_
- reconstruct the URL with the remaining entries

**Comment by ArthurHoaro** [comment]
That's definitely better.

**Comment by virtualtam** [comment]
Should we merge this PR as-is, or update it to improve the URL sanitizing process?

**Comment by nodiscc** [comment]
As you wish; I will not have time to work on your proposed solution right now. I think it can be merged as is. However the bug https://github.com/shaarli/Shaarli/pull/314#issuecomment-128686249 is still present - this can be opened as another issue.

**Comment by virtualtam** [comment]
Let's go for the clean, test-covered way then ;-)

File: application/Url.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ *
+ * Source: http://php.net/manual/en/function.parse-url.php#106731
+ *
+ * @see http://php.net/manual/en/function.parse-url.php
+ *
+ * @param array $parsedUrl an array-represented URL
+ *
+ * @return string the string representation of the URL
+ */
+function unparse_url($parsedUrl)
```
Comment (virtualtam) [inline application/Url.php]: I thought I could avoid this by using http://php.net/manual/en/function.http-build-url.php as it's documented in the PHP doc, but it turns to be a 3rd party PECL extension...

File: application/Url.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    protected $parts;
+
+    /**
+     * Parses a string containing a URL
+     *
+     * @param string $url a string containing a URL
+     */
+    public function __construct($url)
+    {
+        $this->parts = parse_url($url);
```
Comment (virtualtam) [inline application/Url.php]: it's sad PHP has no built-in object representation for URLs / URIs

**Pull request merged**